### PR TITLE
Token Detection not persisting

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
Tokens detected and added to the wallet were not persisting after we kill and re-launch the app

**Proposed Solution**
Working as expected, the tokens are persisting by the network.

**Test Cases**
Case1:
* Import an account with tokens
* Imported the tokens on mainnet
* kill the app
* re-launch the app
* The tokens persisted
Case2:
* Add avalanche network
* Add tokens
* kill the app
* relaunch the app
* should open on avalanche network with tokens persisted

**Screenshots/Recordings**
https://recordit.co/m3XogMlJw8

**Issue**

Progresses #5576 I truly believe this one it's already been fixed since doesn't happen to me on this PR

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by gantunesr — COMMENTED** [review]
Is this a new bug? When was it introduced?

**Review by jpcloureiro — APPROVED** [review]
code lgtm

~~merging main should resolve audit errors~~
EDIT:  just noticed this PR does not target main but `release/5.14.0`
Cherry picking commits seem to be appropriate

File: patches/@metamask+assets-controllers+1.0.1.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/@metamask/assets-controllers/dist/TokensController.js b/node_modules/@metamask/assets-controllers/dist/TokensController.js
+index 4a18560..4297ef3 100644
+--- a/node_modules/@metamask/assets-controllers/dist/TokensController.js
++++ b/node_modules/@metamask/assets-controllers/dist/TokensController.js
+@@ -528,7 +528,7 @@ class TokensController extends base_controller_1.BaseController {
+             newAllIgnoredTokens = Object.assign(Object.assign({}, allIgnoredTokens), { [chainIdToAddTokens]: newIgnoredNetworkTokens });
+         }
+         let newAllDetectedTokens = allDetectedTokens;
+-        if (newDetectedTokens === null || newDetectedTokens === void 0 ? void 0 : newDetectedTokens.length) {
++        if (newDetectedTokens === null || newDetectedTokens === void 0 ? void 0 : !!newDetectedTokens) {
```
[the remark this answers is not in the document]
Comment (tommasini) [reply patches/@metamask+assets-controllers+1.0.1.patch:10]: Not yet, but there will be 🙏

It was introduced on version 1.0.1 of assets-controlller https://github.com/MetaMask/core/blob/main/packages/assets-controllers/CHANGELOG.md

**Comment by tommasini** [comment]
Add the note that version 3.0.1 of assets-controller, solves this issue and we can eliminate this package once we update to it

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**